compiler/rustc_target/src/spec/targets/x86_64_pc_windows_msvc.rs

```diff
@@ -3,6 +3,7 @@ use crate::spec::{base, SanitizerSet, Target};
 pub fn target() -> Target {
     let mut base = base::windows_msvc::opts();
     base.cpu = "x86-64".into();
+    base.features = "+cmpxchg16b,+sse3".into();
```

Comment:
`+sahf` too?

Reply:
That I am not sure. x86-64-v2 onwards don't use sahf though it is listed in the features. CMPXCHG16B and SSE3 are listed. Does that mean Rustc doesn't yet make use of these instructions?

Reply:
Not sure what you are seeing but x86-64-v2 totally does specify SAHF:
https://github.com/llvm/llvm-project/blob/a9700904765590ca2fbf08c0cc36d0da1107d3a7/llvm/lib/Target/X86/X86.td#L808-L809

Reply:
I used this
```
rustc --print=cfg -C target-cpu=x86-64-v2
debug_assertions
panic="unwind"
target_arch="x86_64"
target_endian="little"
target_env="msvc"
target_family="windows"
target_feature="cmpxchg16b"
target_feature="fxsr"
target_feature="popcnt"
target_feature="sse"
target_feature="sse2"
target_feature="sse3"
target_feature="sse4.1"
target_feature="sse4.2"
target_feature="ssse3"
target_has_atomic="16"
target_has_atomic="32"
target_has_atomic="64"
target_has_atomic="8"
target_has_atomic="ptr"
target_os="windows"
target_pointer_width="64"
target_vendor="pc"
windows
```

Did some more digging and turns out SAHF seems to be for kernel rather than userspace code. Hence it is not listed in the rustc command I ran above.

Reply:
It also seems to be tied to virtualization and for kernel space.

Reply:
Ah seems like LLVM does indeed explicitly disable it for 64-bit targets: https://github.com/llvm/llvm-project/blob/7ddc32052546abd41656d2e670f3902b1bf805a7/llvm/lib/Target/X86/X86Subtarget.h#L228C40-L228C53

Reply:
Ok, I've been researching this a bit more. On balance I do think we should add `+sahf`. While I don't think it's that important, it was seen fit to be included in [x86-64-v2](https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels) feature level and I think we should align ourselves with that (although obviously dropping the SSE level back down to 3).

Reply:
done!